compiler/src/dotty/tools/dotc/ast/TreeInfo.scala

```diff
@@ -512,8 +512,8 @@ trait TypedTreeInfo extends TreeInfo[Type] { self: Trees.Instance[Type] =>
       val (meth, targs, argss) = decomposeCall(fn)
       (meth, targs, argss :+ args)
     case TypeApply(fn, targs) =>
-      val (meth, Nil, Nil) = decomposeCall(fn)
-      (meth, targs, Nil)
+      val (meth, targss, args) = decomposeCall(fn)
```

Comment:
I don't understand the change. Can you give an example where this matters?

Reply:
The test added in the same commit (e199fc838e8c437c6cf88b7f27bbb6f350ed3bb2) breaks with a match error before this change